frontend/talentsearch/src/js/components/governmentInfoForm/GovernmentInfoForm.test.tsx

```diff
@@ -0,0 +1,69 @@
+/**
+ * @jest-environment jsdom
+ */
+import "@testing-library/jest-dom";
+import { render, screen, fireEvent, act } from "@testing-library/react";
+import React from "react";
+import { IntlProvider, MessageFormatElement, useIntl } from "react-intl";
+import GovInfoFormContainer from "./GovernmentInfoForm";
+
+const renderWithReactIntl = (
+  component: React.ReactNode,
+  locale?: "en" | "fr",
+  messages?: Record<string, string> | Record<string, MessageFormatElement[]>,
+) => {
+  return (
+    <IntlProvider locale={locale || "en"} messages={messages}>
+      {component}
+    </IntlProvider>
+  );
+};
+
+let container: HTMLDivElement;
+beforeEach(() => {
+  // setup a DOM element as a render target
+  container = document.createElement("div");
+  document.body.appendChild(container);
+});
+
+jest.setTimeout(30000);
+
+test("Test form display rendering", async () => {
+  // timeout for hopefully things to load first then test?
+  act(() => {
+    render(renderWithReactIntl(<GovInfoFormContainer />));
+  });
+  // eslint-disable-next-line no-promise-executor-return
+  await new Promise((r) => setTimeout(r, 10000));
+
+  const button = screen.getByText("Yes, I am a Government of Canada employee");
```

Comment:
I was thinking getByTestId is last thing to choose, since no element has that attribute as default. So do we need to add data-testid to all the element we need to test?

Reply:
I agree with @RVany .  The RTL guide suggests using that selector as a last resort: and https://testing-library.com/docs/queries/about/#priority .

Reply:
Fair enough. 👍

Reply:
Ah interesting, in cypress land, `data-testid` [is very high on recommendations.](https://docs.cypress.io/guides/references/best-practices#:~:text=When%20determining%20an%20unique%20selector%20it%20will%20automatically%20prefer%20elements%20with%3A)

Not a blocker though :) I'll reflect on this divergence, as it will eventually affect our writing e2e cypress tests. (and our selector strategy will fragment, which is maybe ok.)

Reply:
Can we mark the as resolved?

Reply:
Interesting! Two libraries with conflicting recommendations. I'll mark this as resolved but we should table this for a future discussion.